test/new_relic/agent/transaction_test.rb

```diff
@@ -339,6 +342,49 @@ def test_apdex_success_with_ignored_error
       )
     end
 
+    def test_noticed_errors_that_are_expected_do_not_impact_apdex
+      txn_name = 'Controller/BigDweebEnergy'
+      # this one doesn't impact Apdex
+      in_web_transaction(txn_name) do
+        Transaction.notice_error(RuntimeError.new, expected: true)
+      end
+
+      # this one does impact Apdex
+      in_web_transaction(txn_name) do
+        Transaction.notice_error(RuntimeError.new)
+      end
+
+      assert_metrics_recorded(
+        txn_name => {},
+        'Apdex' => {:apdex_s => 1, :apdex_t => 0, :apdex_f => 1},
+        'Apdex/BigDweebEnergy' => {:apdex_s => 1, :apdex_t => 0, :apdex_f => 1}
```

Comment:
I'm not very familiar with this part of the codebase. If both incorrectly impacted Apdex, what would these scores look like?

Reply:
I borrowed this pattern from the ignored errors tests in which one is ok and one is bad.

Given the 2 transactions, one should good (`apdex_s = 1`) and one should be bad (`apdex_f = 1`). If both errors impacted apdex, none would be good (`apdex_s = 0`) and both would be bad (`apdex_f = 2`).

This particular test which uses (`expected: true`) should pass with what we have now in dev. We just unfortunately didn't previously have any test coverage for the behavior. And then the new tests below this point should fail in dev and require the changes delivered by this PR to pass.

Reply:
Thank you!